kanidmd/src/lib/idm/oauth2.rs

```diff
@@ -515,29 +515,32 @@ impl Oauth2ResourceServersReadTransaction {
         // We are now GOOD TO GO! 🥳
 
         // Later we need to limit this and use refresh tokens.
-        let iat = (OffsetDateTime::unix_epoch() + ct).unix_timestamp();
-        let exp = code_xchg.uat.expiry.unix_timestamp();
+        let issued_at = (OffsetDateTime::unix_epoch() + ct).unix_timestamp();
+        let expiry = code_xchg.uat.expiry.unix_timestamp();
 
         // Used in the access token response.
-let expires_in = exp.checked_sub(iat).ok_or_else(|| {
-    lsecurity!(
-        audit,
-        "User Auth Token has expired before we could publish the oauth2 response"
-    )
-    Oauth2Error::AccessDenied
-)?;
+        let expires_in = expiry
+            .checked_sub(issued_at)
+            .and_then(|r| if r.is_negative() { None } else { Some(r) })
```

Comment:
I'm still confused about what type `expiry` and `issued_at` are. If they're signed types (which seemed implied to `is_negative`, then the checked sub is kind of pointless. I kinda suggested the change assuming they were unsigned.

Reply:
They have to be i64 because lol js standards dont seem to have unsigned types :( 

The cheked sub is probably still a good idea incase something wild happens, and the resulting code is pretty clean anyway, so it's a lot of muchness I think